nanoFramework.CoreLibrary/System/Reflection/CustomAttributesHelpers.cs

```diff
@@ -12,30 +12,42 @@ public static object[] GetCustomAttributesInternal(object[] rawAttributes)
             // get the custom attributes data for the field
             // these are returned "encoded" in an object array with 2 positions for each attribute
             // 1st the attribute type
-            // 2nd the constructor parameter or null, if the attribute has no constructor
+            // 2nd the constructor parameter or null, if the attribute has no constructor parameters
             // 
             // current limitations: 
             // - works only for constructors with a single parameter
             // - the parameter has to be a string or numeric type
             //
             // both limitations above can be relatively easily overcome by adding the appropriate code at the native handler
 
-            object[] attributes = new object[rawAttributes.Length / 2];
+            if (rawAttributes is null)
```

Comment:
We need to make this super clear for future reference. 
Can you please add a comment on each situation to make it abundantly clear what is happening/being covered in each step/block?

Reply:
The code should be a lot clearer now. Let me know.